subprojects/core/src/main/java/org/gradle/execution/plan/OrdinalNode.java

```diff
@@ -116,4 +119,10 @@ public Type getType() {
     public int getOrdinal() {
         return ordinal;
     }
+
+    public void addDependenciesFrom(TaskNode taskNode) {
+        // Only add hard successors that will actually be executed
+        Stream<Node> executingSuccessors = stream(taskNode.getHardSuccessors().spliterator(), false).filter(Node::isRequired);
```

Comment:
Let's keep an eye on this in perf tests; do we have any alternatives to opening a stream?

Reply:
I'd be shocked if there was a noticeable impact.  A loop would likely be slightly more performant, but it's only doing this once per task in the graph.  There is most certainly overhead associated with it, but my gut feeling is that it's not significant.  We'll see if the perf tests say otherwise.